Incorporate uglify's new object property mangling capabilities

jrhite opened the pull request:
Updated grunt wrapper to be able to hook into uglify's new object property name mangling functionality, the ability to pass files containing property and variable name exceptions and the ability to use a cache to coordinate symbol mangling across multiple calls to uglify.

See uglifyjs v2.4.19 https://www.npmjs.com/package/uglify-js for more info.

vladikoff commented:
Thanks for the PR! some tests are failing though :(

jrhite commented:
The 4 failing tests have been failing since commit: 8f9bb92. Earlier comments from @XhmikosR say that he is on it and fixing them.

I added 5 new tests specific to the new functionality and they're all passing.

XhmikosR commented:
I'm still waiting for a reply from @vladikoff 'cause as you can see from here https://github.com/gruntjs/grunt-contrib-uglify/compare/tests, I only managed to fix 2...

pdachtera commented:
+1 for the PR!

vladikoff commented:
@jrhite could you please rebase your PR? also no need to update the changelog

jrhite commented:
Rebased PR and reverted changelog changes...

Thanks!


On Tue, Mar 31, 2015 at 3:33 AM, Vlad Filippov <notifications@github.com>
wrote:

> @jrhite <https://github.com/jrhite> could you please rebase your PR? also
> no need to update the changelog
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gruntjs/grunt-contrib-uglify/pull/312#issuecomment-87824130>
> .
>

XhmikosR commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "grunt-contrib-uglify",
   "description": "Minify files with UglifyJS.",
-  "version": "0.8.1",
+  "version": "0.9.0",
```
This should be reverted too.

chesserik commented:
+1 !

jk- commented:
really good idea! +1

yneves commented:
best idea ever +1

jrhite commented:
@XhmikosR I just reverted the package.json version change and rebased to the latest code. Should be good to go.

Cheers!

XhmikosR commented on docs/uglify-options.md:
```diff
@@ -139,3 +139,31 @@ Type: `Boolean`
 Pass this flag if you don't care about full compliance with Internet Explorer 6-8 quirks.
+
+## mangleProperties
+Type: `Boolean`
```
Needs 2 tailing spaces for a new line for all the newly added types.

jrhite commented:
@XhmikosR good eye again! (uggh...MD requires 2 extra spaces like that...still learning it :-) )

I went ahead and add the 2 trailing spaces to all the new options in docs/uglify-options.md. I also fixed some of the older options that were missing this too.

Checked in and rebased. Thanks!

XhmikosR commented:
Thanks!

jrhite commented:
Thanks!

@XhmikosR when/how will this make it to the npm repo? it will be version 0.8.2, correct?

XhmikosR commented:
I don't make the releases myself so when @vladikoff thinks it's OK :P

Though, perhaps we could wait a couple of days in case we merge some of the other PRs too.

jrhite commented:
Ah, got it ok. Thanks...

No big rush here, but the sooner the better of course! :-)


On Thu, Apr 2, 2015 at 12:28 PM, XhmikosR <notifications@github.com> wrote:

> I don't make the releases myself so when @vladikoff
> <https://github.com/vladikoff> thinks it's OK :P
>
> Though, perhaps we could wait a couple of days in case we merge some of
> the other PRs too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gruntjs/grunt-contrib-uglify/pull/312#issuecomment-88740781>
> .
>
